### DISCUSSION TITLE
Why is the binding to a string[] resulting in an empty value?

### DISCUSSION THREAD

**Discussion by IronEaterBeast** [AvaloniaUI/Avalonia#14405] [Q&A]
`<DataGrid Margin="20" ItemsSource="{Binding urlList}">
	<DataGrid.Columns>
		<DataGridTextColumn Header="Url" Width="2*"
		   Binding="{Binding [0]}" />
		<DataGridTextColumn Header="Url2" Width="2*"
		   Binding="{Binding [1]}" />
	</DataGrid.Columns>
</DataGrid>`

The data structure of urlList is an ObservableCollection<string[]>.

**Comment by timunie** [comment]
try ReflectionBinding

**Reply by IronEaterBeast** [reply]
Does ReflectionBinding support AOT? My program requires AOT support.

**Reply by timunie** [reply]
AOT and DataGrid is a nightmare tbh. You need to tell the trimmer what to keep in order to make it work... #14335

**Reply by IronEaterBeast** [reply]
Alright, I understand. But I thought <DataGridTextColumn Header="Url" Binding="{Binding [0]}" /> should work. Is string[] not supported?

**Reply by timunie** [reply]
works for me in debug more, so must be your setup somehow or AOT thing as I said earlier.
